api/main.py

```
@@ -600,9 +605,32 @@ async def get_streams(
 
 @app.post("/encrypt-user-data", tags=["user_data"])
 @wrappers.rate_limit(30, 60 * 5, "user_data")
-async def encrypt_user_data(user_data: schemas.UserData):
+async def encrypt_user_data(user_data: schemas.UserData, request: Request):
+    if not settings.is_public_instance:
+        # validate API Password
+        if (
+            not user_data.api_password
+            or user_data.api_password != settings.api_password
+        ):
```

Ruff

Use a single `if` statement instead of nested `if` statements

(SIM102)